packages/contractkit/src/contract-cache.ts

```diff
@@ -0,0 +1,104 @@
+import { CeloContract } from 'src/base'
+import { ContractKit } from 'src/kit'
+import { BondedDepositsWrapper } from 'src/wrappers/BondedDeposits'
+import { ExchangeWrapper } from 'src/wrappers/Exchange'
+import { GoldTokenWrapper } from 'src/wrappers/GoldTokenWrapper'
+import { StableTokenWrapper } from 'src/wrappers/StableTokenWrapper'
+import { ValidatorsWrapper } from 'src/wrappers/Validators'
+
+const WrapperFactories = {
+  // [CeloContract.Attestations]: AttestationsWrapper,
+  [CeloContract.BondedDeposits]: BondedDepositsWrapper,
+  // [CeloContract.Escrow]: EscrowWrapper,
+  [CeloContract.Exchange]: ExchangeWrapper,
+  // [CeloContract.GasCurrencyWhitelist]: GasCurrencyWhitelistWrapper,
+  // [CeloContract.GasPriceMinimum]: GasPriceMinimumWrapper,
+  [CeloContract.GoldToken]: GoldTokenWrapper,
+  // [CeloContract.Governance]: GovernanceWrapper,
+  // [CeloContract.MultiSig]: MultiSigWrapper,
+  // [CeloContract.Random]: RandomWrapper,
+  // [CeloContract.Registry]: RegistryWrapper,
+  // [CeloContract.Reserve]: ReserveWrapper,
+  // [CeloContract.SortedOracles]: SortedOraclesWrapper,
+  [CeloContract.StableToken]: StableTokenWrapper,
+  [CeloContract.Validators]: ValidatorsWrapper,
+}
+
+type CFType = typeof WrapperFactories
+
+interface WrapperCacheMap {
+  // [CeloContract.Attestations]?: AttestationsWrapper,
+  [CeloContract.BondedDeposits]?: BondedDepositsWrapper
+  // [CeloContract.Escrow]?: EscrowWrapper,
+  [CeloContract.Exchange]?: ExchangeWrapper
+  // [CeloContract.GasCurrencyWhitelist]?: GasCurrencyWhitelistWrapper,
+  // [CeloContract.GasPriceMinimum]?: GasPriceMinimumWrapper,
+  [CeloContract.GoldToken]?: GoldTokenWrapper
+  // [CeloContract.Governance]?: GovernanceWrapper,
+  // [CeloContract.MultiSig]?: MultiSigWrapper,
+  // [CeloContract.Random]?: RandomWrapper,
+  // [CeloContract.Registry]?: RegistryWrapper,
+  // [CeloContract.Reserve]?: ReserveWrapper,
+  // [CeloContract.SortedOracles]?: SortedOraclesWrapper,
+  [CeloContract.StableToken]?: StableTokenWrapper
+  [CeloContract.Validators]?: ValidatorsWrapper
+}
+
+export class WrapperCache {
+  // private wrapperCache: Map<CeloContract, any> = new Map()
+  private wrapperCache: WrapperCacheMap = {}
+
+  constructor(readonly kit: ContractKit) {}
+
+  // getAttestations() {
+  //   return this.getWrapper(CeloContract.Attestations, newAttestations)
+  // }
+  getBondedDeposits() {
+    return this.getContract(CeloContract.BondedDeposits)
```

Comment:
Will you add types these? I see you set up the template below in `getContract` but aren't using it yet

Reply:
do you refer to specify the return types explicitly?
All methods are typed, and have concrete types, not `any` or `ts-ignores`